### ISSUE TITLE
Extended Codepoints in Alternative Material Design Icon Font

### ISSUE THREAD

**Issue by michaelquigley** [ocornut/imgui#3247]
**Version/Branch of Dear ImGui:**

Version: 1.76

**Back-end/Renderer/Compiler/OS**

Back-ends: `imgui_impl_glfw.cpp` + `imgui_impl_opengl3.cpp`
Compiler: MSVC 19.25.28614
Operating System: Windows 10

**My Issue/Question:**

I am trying to use an icon font with fairly large (5 digit) codepoints from my application. An overview of the contents of the font is here:

https://cdn.materialdesignicons.com/5.2.45/

My application is working fine with the standard material design icon font, which uses 4 digit codepoints.

I am merging the material design font into my font atlas like this:

```
static const ImWchar icons_ranges[] = { 0xF000, 0xFFFF, 0 };
ImFontConfig icons_config; icons_config.MergeMode = true; icons_config.PixelSnapH = true;
auto materialdesign = new unsigned char[fonts::materialdesign_length];
memcpy(materialdesign, fonts::materialdesign, size_t(fonts::materialdesign_length));
ImGui::GetIO().Fonts->AddFontFromMemoryTTF(materialdesign, fonts::materialdesign_length, font_size, &icons_config, icons_ranges);
```

The first issue is that the `icons_ranges` does not take a large enough value for the codepoints in my font. I tried also passing `nullptr` to `AddFontFromMemoryTTF` to see if it would just merge everything, but had no success there.

I am trying to reference the glyphs in the font using UTF-8 bytes like this:

```
ImGui::TreeNodeEx(fmt::format("{0} {1}", "\xef\x85\x87\x64", n->name()).c_str(), leaf_flags);
```

...or by letting the compiler do the conversion, like this:

```
ImGui::TreeNodeEx(fmt::format("{0} {1}", u8"\uf147d", n->name()).c_str(), leaf_flags);
```

No success with either approach. Any help would be appreciated. Definitely not a unicode expert, but did my best to try and answer my own questions as much as possible before submitting a new issue.

**Comment by ocornut** [comment]
Hello

You’ll need to enable IMGUI_USE_WCHAR32 in imconfig.h to use unicode codepoint larger than 0xFFFF.

**Comment by michaelquigley** [comment]
Finally managed to get this working. Once I got the `IMGUI_USE_WCHAR32` happening, I was able to use:

`static const ImWchar icons_ranges[] = { 0xF1000, 0xFF000, 0 };`

And then I had to figure out the magic incantation to make the compiler happy with the extended codepoints. This seems to work:

`ImGui::TreeNodeEx(u8"\U000F147D" "Hello", leaf_flags);`

Thank you for the help! Might be a good FAQ addition?
